### PR TITLE
DT-1338: 🎨 Add roles endpoint

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/UserService.java [hunk 1]
```diff
@@ -157,8 +161,13 @@ public void removeUsersAccessRoleForCaseload(final String username, final String
         if (!userRepository.isRoleAssigned(username, caseload, role.getRoleId())) {
             throw EntityNotFoundException.withMessage("Role [%s] not assigned to user [%s] at caseload [%s]", roleCode, username, caseload);
         }
-        userRepository.removeRole(username, caseload, role.getRoleId()); // Don't care if it doesn't exist...
-        log.info("Removed role '{}' from username '{}' at caseload '{}'", roleCode, username, caseload);
+        if (userRepository.removeRole(username, caseload, role.getRoleId()) > 0) {
+            telemetryClient.trackEvent(
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/UserService.java:165]: add telemetry for adding and removing roles

File: src/test/java/uk/gov/justice/hmpps/prison/service/GlobalSearchServiceImplTest.java [hunk 2]
```diff
@@ -71,7 +70,7 @@ public void testFindOffendersByOffenderNo() {
 
         service.findOffenders(criteria, pageRequest);
 
-        verify(inmateRepository, Mockito.times(1)).findOffenders(eq(TEST_OFFENDER_NO_QUERY), any(PageRequest.class));
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmpps/prison/service/GlobalSearchServiceImplTest.java:74]: default is `times(1)` anyway

File: src/test/java/uk/gov/justice/hmpps/prison/service/GlobalSearchServiceImplTest.java [hunk 3]
```diff
@@ -135,8 +134,8 @@ public void testFindOffendersWithPncNumberMatch() {
 
         assertThat(response.getItems()).isNotEmpty();
 
-        Mockito.verifyZeroInteractions(inmateRepository);
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmpps/prison/service/GlobalSearchServiceImplTest.java:138]: `verifyZeroInteractions` is deprecated

File: src/test/java/uk/gov/justice/hmpps/prison/service/UserServiceImplTest.java [hunk 4]
```diff
@@ -202,15 +254,4 @@ public void testGetOffenderCategorisationsBatching() {
 
         verify(userRepository, times(2)).getUserListByUsernames(anyList());
     }
-
-    private Page<UserDetail> pageResponse(final int userCount) {
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmpps/prison/service/UserServiceImplTest.java:206]: not used in this class

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**